### ISSUE TITLE
mutate jobs via RunnerInput

### ISSUE THREAD

**Issue by mmjconolly** [sifive/wake#1374]
When a job has a `Plan` that specifies an `Environment` that env is put in the wake database and the user can see it.
If a `Runner` wants to exec out to something that will truly execute the plan, it sometimes wants to add to the environment (via the runner's `pre` def).
The wake database doesn't have a way to track theses additions.

**Comment by JakeSiFive** [comment]
Nope, nothing from pre ever makes it into the database. The flow works like this

1) A job is created (but not launched), the job is created with command line, environmment, etc... as part of the job
2) The job and its associated "runner input" is passed to a runner
3) The runner eventually (possibly by successive altered RunnerInputs being passed) calls launch with the actual command, environment, etc... but the database never records that information
4) After the job is done the RunnerOutput is given to the runner where the runner is allowed to return a modified RunnerOutput from there (via say, a series of wrapped posts)

We should probably record the launch parameters in the database however
